panther_gpiod/include/panther_gpiod/gpio_driver.hpp

```diff
@@ -0,0 +1,265 @@
+   *
+   * @param pin GPIOpin to check.
+   * @return True if the pin is active, false otherwise.
+   */
+  bool is_pin_active(GPIOpin pin);
+
+  /**
+   * @brief Sets the value for a specific GPIO pin.
+   *
+   * @param pin GPIOpin to set the value for.
+   * @param value New value to set for the pin.
+   * @return True if the value is set successfully, false otherwise.
+   */
+  bool set_pin_value(GPIOpin pin, bool value);
+
+  /**
+   * @brief Changes the direction of a specific GPIO pin.
+   *
+   * @param pin GPIOpin to change the direction for.
+   * @param direction New direction for the pin.
+   * @throws std::runtime_error if there is an error while setting the GPIO pin value.
+   */
+  void change_pin_direction(GPIOpin pin, gpiod::line::direction direction);
+
+  /**
+   * @brief Enables asynchronous monitoring of GPIO pin events.
+   */
+  void gpio_monitor_on();
+
+  /**
+   * @brief Disables asynchronous monitoring of GPIO pin events.
+   */
+  void gpio_monitor_off();
+
+private:
+  std::unique_ptr<gpiod::line_request> create_line_request(gpiod::chip & chip, const GPIOpin pin);
+  std::unique_ptr<gpiod::line_request> create_line_request(
+    gpiod::chip & chip, const std::vector<GPIOpin> & pins);
+  gpiod::line_settings generate_line_settings(const GPIOinfo & pin_info);
+  GPIOpin get_pin_from_offset(gpiod::line::offset offset) const;
+  GPIOinfo & get_pin_info_ref(GPIOpin pin);
+  void configure_line_request(gpiod::chip & chip, gpiod::request_builder & builder, GPIOpin pin);
+  void monitor_async_events();
+  void handle_edge_event(const gpiod::edge_event & event);
+  bool is_gpio_monitor_thread_running() const;
+
+  /**
+   * @brief Callback function for GPIO edge events.
+   *
+   * @param gpio_info Information related to the state of the GPIO pin for which the event took
+   * place.
+   */
+  std::function<void(const GPIOinfo & gpio_info)> gpio_edge_event_callback;
+
+  /**
+   * @brief Mapping of GPIO pins to their respective names.
+   *
+   * This map contains the names associated with different GPIO pins.
+   */
+  const std::map<GPIOpin, std::string> pin_names_{
+    {GPIOpin::WATCHDOG, "WATCHDOG"},
+    {GPIOpin::AUX_PW_EN, "AUX_PW_EN"},
+    {GPIOpin::CHRG_DISABLE, "CHRG_DISABLE"},
+    {GPIOpin::CHRG_SENSE, "CHRG_SENSE"},
+    {GPIOpin::DRIVER_EN, "DRIVER_EN"},
+    {GPIOpin::E_STOP_RESET, "E_STOP_RESET"},
+    {GPIOpin::FAN_SW, "FAN_SW"},
+    {GPIOpin::GPOUT1, "GPOUT1"},
+    {GPIOpin::GPOUT2, "GPOUT2"},
+    {GPIOpin::GPIN1, "GPIN1"},
+    {GPIOpin::GPIN2, "GPIN2"},
+    {GPIOpin::LED_SBC_SEL, "LED_SBC_SEL"},
+    {GPIOpin::SHDN_INIT, "SHDN_INIT"},
+    {GPIOpin::VDIG_OFF, "VDIG_OFF"},
+    {GPIOpin::VMOT_ON, "VMOT_ON"},
+  };
+
+  /**
+   * @brief Vector containing GPIO pin configuration information.
+   *
+   * This vector stores information related to GPIO pins such as pin configuration,
+   * direction, value, etc.
+   */
+  std::vector<GPIOinfo> gpio_info_;
```

Comment:
```suggestion
  std::vector<GPIOinfo> gpio_info_collection_;
```
gpio_info suggest that this is single instance of GPIOinfo class. Can we use more descriptive name like collection or storage?